src/main/core/scheduler/pools/bounded.rs

```diff
@@ -175,6 +175,25 @@ impl ParallelismBoundedThreadPool {
 
     /// Create a new scope for the pool. The scope will ensure that any task run on the pool within
     /// this scope has completed before leaving the scope.
+    //
+    // SAFETY: This works because:
+    //
+    // 1. WorkerScope<'scope> is covariant over 'scope.
```

Comment:
This is probably an embarrassing gap in my knowledge, but could you tell me or point me to definitions of covariant, invariant, and contravariant in this context?